flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/data/TestRowProjection.java

```diff
@@ -61,52 +63,63 @@ private RowData writeAndRead(String desc, Schema writeSchema, Schema readSchema,
       appender.add(row);
     }
 
-    Iterable<RowData> records =
+    Avro.ReadBuilder builder =
         Avro.read(Files.localInput(file))
             .project(readSchema)
-            .createResolvingReader(FlinkPlannedAvroReader::create)
-            .build();
+            .createReaderFunc(FlinkAvroReader::new);
+    if (useAvroPlannedReader) {
+      builder =
+          Avro.read(Files.localInput(file))
+              .project(readSchema)
+              .createResolvingReader(FlinkPlannedAvroReader::create);
+    }
+    Iterable<RowData> records = builder.build();
 
     return Iterables.getOnlyElement(records);
   }
 
-  @Test
-  public void testFullProjection() throws Exception {
+  @ParameterizedTest
+  @ValueSource(booleans = {false, true})
```

Comment:
Do we add this to all of the tests? Maybe use `@TestTemplate` for this

Reply:
Let me check, but I think `@TestTemplate` is an option.

Reply:
Actually, after reviewing this part, I'm not convinced `@TestTemplate` is much better: we would need to add `@TestTemplate` and `@ExtendWith()` annotations. So, still two annotations to add + the context provider.

@pvary thoughts ?

Reply:
`@TestTemplate` is for parameterization at the class level, which seems exactly what's being done here. So rather than adding `@ValueSource` to each single method it makes sense to do this at the class level and annotate all tests with `@TestTemplate`

Reply:
OK, let me do it.

Reply:
@nastra by the way, I was confused about the expectation here because `@TestTemplate` is not necessary at class level, it can be at method level too. Actually both `@ParameterizedTest` and `@RepeatedTest` are based on `@TestTemplate` (they use custom invocation context provider). That's why I was confused (as `@ParameterizedTest` is actually a "specialized" `@TestTemplate`) 😄

Reply:
I update the test to use `@TestTemplate`.